### PR TITLE
IA pour Pacman

### PR DESCRIPTION
**Pull request by hayatbellafkih**
IA pour Pacman

### CODE REVIEW

File: src/main/java/nl/tudelft/jpacman/npc/ghost/Navigation.java [hunk 1]
```diff
@@ -40,35 +40,40 @@ private Navigation() {
...
-			if (s == to) {
-				List<Direction> path = n.getPath();
-				return path;
-			}
-			visited.add(s);
-			addNewTargets(traveller, targets, visited, n, s);
+
+		targets.add(n);
+
+		while (from!=to && !targets.isEmpty()) {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/Navigation.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539284/issues/source?bid=3256381&fileBranchId=3281825#l50)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/clydePursuit.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package nl.tudelft.jpacman.npc.ghost;
+
+/**
+ * Created by UMONS on 18/04/2016.
+ */
+public class clydePursuit {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/clydePursuit.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539289/issues/source?bid=3256381&fileBranchId=3281825#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/blinkyPursuit.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package nl.tudelft.jpacman.npc.ghost;
+
+/**
+ * Created by UMONS on 18/04/2016.
+ */
+public class blinkyPursuit {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/blinkyPursuit.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539288/issues/source?bid=3256381&fileBranchId=3281825#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/pursuitMove.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package nl.tudelft.jpacman.npc.ghost;
+
+/**
+ * Created by UMONS on 18/04/2016.
+ */
+public class pursuitMove {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/pursuitMove.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539292/issues/source?bid=3256381&fileBranchId=3281825#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/PinkyPursuit.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     * @param d
+     * 		The direction currently followed by Pac-Man.
+     *
+     * @param s
+     * 		The Pac-Man current position.
+     *
+     * @return
+     * 		Four squares in front of Pac-Man in his current direction of travel.
+     */
+    private Square fourSquaresAway(Direction d, Square s){
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/PinkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 's'](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539287/issues/source?bid=3256381&fileBranchId=3281825#l46)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/DispersionMove.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        this.directionCounter = -1;
+    }
+
+    @Override
+    public Direction nextMove(){
+        Square origin = getGhost().getSquare();
+        Square home = getGhost().getHome();
+        Direction d = getGhost().randomMove();
+
+        if(origin == home || getGhost().isInMyCorner()) {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/DispersionMove.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539277/issues/source?bid=3256381&fileBranchId=3281825#l48)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+     * @param blinkyPlace
+     * 		The square currently occupied by Blinky.
+     *
+     * @param destination
+     * 		The square at two square in front of Pac-Man.
+     *
+     * @return
+     * 		The next square where to go.
+     */
+    private Direction myPathTo(Square blinkyPlace, Square destination) {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 'destination'](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539283/issues/source?bid=3256381&fileBranchId=3281825#l46)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+     * @param d
+     * 		The direction currently followed by Pac-Man.
+     *
+     * @param s
+     * 		The Pac-Man current position.
+     *
+     * @return
+     * 		Two squares in front of Pac-Man in his current direction of travel.
+     */
+    private Square twoSquaresAway(Direction d, Square s){
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 's'](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539283/issues/source?bid=3256381&fileBranchId=3281825#l74)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/moveStrategy.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package nl.tudelft.jpacman.npc.ghost;
+
+/**
+ * Created by UMONS on 18/04/2016.
+ */
+public interface moveStrategy {
```
Comment (Soupline69) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/moveStrategy.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mikis-millemaci/jpacman-framework/file/2525539290/issues/source?bid=3256381&fileBranchId=3281825#l6)

Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/moveStrategy.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543484/issues/source?bid=3267846&fileBranchId=3281829#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/DispersionMove.java (hunk 6 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/DispersionMove.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543471/issues/source?bid=3267846&fileBranchId=3281829#l48)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java (hunk 8 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 's'](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543477/issues/source?bid=3267846&fileBranchId=3281829#l74)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java (hunk 7 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/InkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 'destination'](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543477/issues/source?bid=3267846&fileBranchId=3281829#l46)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/blinkyPursuit.java (hunk 3 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/blinkyPursuit.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543482/issues/source?bid=3267846&fileBranchId=3281829#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/clydePursuit.java (hunk 2 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/clydePursuit.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543483/issues/source?bid=3267846&fileBranchId=3281829#l6)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/Navigation.java (hunk 1 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/Navigation.java]: Codacy Issue found: [Use equals() to compare object references.](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543478/issues/source?bid=3267846&fileBranchId=3281829#l50)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/PinkyPursuit.java (hunk 5 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/PinkyPursuit.java]: Codacy Issue found: [Avoid reassigning parameters such as 's'](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543481/issues/source?bid=3267846&fileBranchId=3281829#l46)

File: src/main/java/nl/tudelft/jpacman/npc/ghost/pursuitMove.java (hunk 4 above)
Comment (Mathieu1609) [inline src/main/java/nl/tudelft/jpacman/npc/ghost/pursuitMove.java]: Codacy Issue found: [Class names should begin with an uppercase character](https://www.codacy.com/app/mathieu-craeye/jpacman-framework_2/file/2525543486/issues/source?bid=3267846&fileBranchId=3281829#l6)

**Comment by avandeursen** [comment]
Please use pull requests to the public framework only for well documented changes that are useful for all JPacman-Framework users.

**Pull request closed without merging**